app/component/itinerary/LegInfo.js

```diff
@@ -32,9 +33,11 @@ export default function LegInfo(
   const { constantOperationRoutes } = config;
   const shouldLinkToTrip =
     !constantOperationRoutes || !constantOperationRoutes[leg.route.gtfsId];
-  const mode = isCallAgency
+  let mode = isCallAgency
     ? 'call'
     : getRouteMode({ mode: leg.mode, type: leg.route.type });
+  const feedId = getFeedWithoutId(leg.route?.gtfsId);
+  mode = isExternalFeed(feedId, config) ? `${mode}-external` : mode;
   const capacity = getCapacityForLeg(config, leg);
```

Comment:
Should getRouteMode() handle the postfixing (using some new params)? I see these code lines repeating in many places.

Reply:
I added the logic to getRouteMode. I left the old way in one place, because it was unclear whether the change could be applied